### PR TITLE
Chore  add express server

### PR DESCRIPTION
**Pull request by rise-erpelding**
## Description
Add express server with Hello World page using nodemon and babel/node.

## Spec

See Issue: [36](https://sparkbox.atlassian.net/browse/FSA2021-36)

## Validation
- [x] This PR has code changes, and our linters still pass.

### To Validate

1. Make sure all PR Checks have passed.
1. Pull down all related branches.
1. Navigate to branch `chore--add-express-server`
2. Run `npm start` to ensure that http://localhost:3000 shows "Hello World!"

### CODE REVIEW

File: package.json
Comment (jordanleven) [inline package.json:6]: 🏆  Boom - this is perfect and does exactly what we need it to do in order to run `server.js` through the Babel transpiler. Nicely done!

File: package.json
Comment (jordanleven) [inline package.json]: This _probably_ doesn't need to be a production dependency since we're only expecting to run this command locally. You _can_ include it as a production dependency but we're running express server through `nodemon` and using `@babel/node`, which are only development dependencies. 

Note that if you move `express` to a dev dependency, you might have to ignore a linting error in server.js that says it should be a production dependency. In this case, we can override it by telling [Eslint to ignore a specific linting](https://stackoverflow.com/questions/34764287/turning-off-eslint-rule-for-a-specific-file) error in the file.

File: server.js
Comment (jordanleven) [inline server.js]: :chefskiss: Glorious! This works as expected!

One thing to consider about the naming, and feel free to ping me about this because I can imagine that this might come across as confusing. This is a long comment so I'm going to give you the opportunity to read the unabridged or abridged version. Choose your own adventure!

### Unabridged
In ES5, we have two new variable types (in addition to `var`) which are `const` (constants, or immutable variables) and `let` (scoped variables but more importantly, mutable). Big difference between the two, you can redeclare the value of a variable declared with `let` but not `const`.

```
// This is okay.
let myFavoriteColor = 'green'
myFavoriteColor = 'blue'

// This is not.
const myFavoriteColor = 'green';
myFavoriteColor = 'blue'
```

Within constant variables though, we have times where the value of const _is_ variable (like saying `const myFavoriteColor = getMyFavoriteColor()`) and times where it's not (`const myFavoriteColor = 'green'`). Th difference here is that the first example that references `getMyFavoriteColor` is assigned a different value based on the return value of `getMyFavoriteColor`, which can be variable. It means that the value of `myFavoriteColor` _might_ be green, or might be red or blue. It all depends on the value returned from `getMyFavoriteColor`. And saying, "It all depends" is the same as saying "it's variable."

Now let's revisit `const myFavoriteColor = 'green'`. Something is different here. `myFavoriteColor` has an assigned value that isn't variable. It's completely static. These are called static constants, and they have a different naming convention. In many languages (Java, PHP, and yes, [JavaScript](https://google.github.io/styleguide/javascriptguide.xml#Constants)) a common convention is to distinguish an immutable variable from an immutable constant by using upper snake case. Which means my code might look like this:

```
const MY_FAVORITE_COLOR = 'green'
const MY_FIRST_NAME = 'Jordan'
const currentDay = getCurrentDay()

const greeting = `Hello, my name is ${MY_FIRST_NAME}, today is ${currentDay}, and  my favorite color is ${MY_FAVORITE_COLOR}`
```

Note that there are four immutable variables here (that is, four variables have their value declared once and never change), but only two are statically constant. `MY_FAVORITE_COLOR` and `MY_FIRST_NAME` will never change, but the value of `currentDay` is variable based on the current day which means that, by definition, `greeting` is also variable.

None of this is required to move this forward, but that's just a convention to be aware of if you're interested.


### Abridged
Consider changing `port` to `PORT`.

**Review by jordanleven — COMMENTED** [review]
Looking good, @rise-erpelding! I left a couple of comments throughout the PR. Let me know what you think!
giphy copy

**Review by FlashGooden — APPROVED** [review]
Tests pass, package.json updated, Server has new ES6 syntax

File: package.json
Comment (jordanleven) [inline package.json:37]: @mcksheridan  - quick question on the movement of this dependency: did you run `npm rm express && npm i express -D`? It looks like the dependency was changed in `package.json` but there should be a corresponding change in `package-lock.json`.

**Review by jordanleven — CHANGES_REQUESTED** [review]
@mcksheridan  - Thanks for making this change! I just put a quick question in about the dependency update.

File: package.json
Comment (mcksheridan) [reply package.json:37]: Completely forgot to run `npm rm express` and went straight for `npm i express -D` instead. I'll fix that!

Comment (jordanleven) [reply package.json:37]: Oh, if you ran that command that should have been fine. I was just making sure that you didn't just copy the text into the dev dependencies section. As long as users pulling down your branch can run `rm -rf node_modules && npm i` without having `package-lock.json` change, you're good to go!

**Review by mcksheridan — APPROVED** [review]

File: package.json
Comment (mcksheridan) [reply package.json:37]: @jordanleven Before we go ahead and merge this in, the way the `package.json` and `package-lock.json` files look now is OK, right? (We're still seeing that merging is blocked until requested changes are addressed.)

**Review by jordanleven — COMMENTED** [review]
Looks great!

**Review by jordanleven — APPROVED** [review]
It looks like I _do_ have to update my review in order to make way for merging. 🤦

**Pull request merged**
